Sources/SwiftSyntax/SyntaxText.swift

```diff
@@ -100,7 +100,10 @@ public struct SyntaxText {
     guard !self.isEmpty && !other.isEmpty else {
       return self.isEmpty && other.isEmpty
     }
-    return (other.baseAddress! <= self.baseAddress! && self.baseAddress! + count <= other.baseAddress! + other.count)
+    let doesOtherIncludeSelfBaseAddress: Bool = other.baseAddress! <= self.baseAddress!
+    let sendEndBound = UnsafePointer<UInt8>(self.baseAddress! + count)
+    let otherEndBound = UnsafePointer<UInt8>(other.baseAddress! + other.count)
+    return (doesOtherIncludeSelfBaseAddress && sendEndBound <= otherEndBound)
```

Comment:
Out of curiosity: Is there a compile-time difference between this and 

```suggestion
    return (other.baseAddress! <= self.baseAddress! && sendEndBound <= otherEndBound)
```

Reply:
Great suggestion, I have measured and the result is `7.48 ms`. Removed the variable for the first condition.